### PR TITLE
Add tongyi tts&tts function optimization

### PR DESCRIPTION
**Pull request by charli117**

### CODE REVIEW

File: api/core/model_runtime/model_providers/tongyi/tts/tts.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+from core.model_runtime.model_providers.__base.tts_model import TTSModel
+from core.model_runtime.model_providers.openai._common import _CommonOpenAI
+
+from flask import Response, stream_with_context
+import dashscope
+from dashscope.audio.tts import SpeechSynthesizer
+import concurrent.futures
+
+
+class TongyiText2SpeechModel(_CommonOpenAI, TTSModel):
```
Comment (takatost) [inline api/core/model_runtime/model_providers/tongyi/tts/tts.py]: I think `_CommonOpenAI` might not be compatible with Tongyi's tts

File: api/core/model_runtime/model_providers/openai/tts/tts.py [hunk 2]
```diff
@@ -1,30 +1,23 @@
...
 
-from typing_extensions import Literal
+import dashscope
 from flask import Response, stream_with_context
-from openai import OpenAI
 import concurrent.futures
 
 
-class OpenAIText2SpeechModel(_CommonOpenAI, TTSModel):
+class TongyiText2SpeechModel(_CommonOpenAI, TTSModel):
```
Comment (takatost) [inline api/core/model_runtime/model_providers/openai/tts/tts.py]: Why did OpenAI's Class name change to Tongyi? 🤔

File: api/core/model_runtime/model_providers/openai/tts/tts.py [hunk 3]
```diff
@@ -116,128 +108,35 @@ def _tts_invoke_streaming(self, model: str, credentials: dict, content_text: str
         :return: text translated to audio file
         """
         # transform credentials to kwargs for model instance
-        credentials_kwargs = self._to_credential_kwargs(credentials)
+        dashscope.api_key = credentials.get('dashscope_api_key')
```
Comment (takatost) [inline api/core/model_runtime/model_providers/openai/tts/tts.py]: Maybe you changed the wrong place?

**Review by takatost — CHANGES_REQUESTED** [review]

File: api/core/model_runtime/model_providers/openai/tts/tts.py (hunk 2 above)
Comment (charli117) [reply api/core/model_runtime/model_providers/openai/tts/tts.py]: > Why did OpenAI's Class name change to Tongyi? 🤔

@takatost 😅Impatient, produced a lot of bugs, has been fixed

File: api/core/model_runtime/model_providers/tongyi/_common.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from core.model_runtime.errors.invoke import InvokeError
+
+
+class _CommonTongyiOpenAI:
```
Comment (takatost) [inline api/core/model_runtime/model_providers/tongyi/_common.py]: weird name...

**Review by takatost — CHANGES_REQUESTED** [review]

File: api/core/model_runtime/model_providers/tongyi/_common.py (hunk 4 above)
Comment (charli117) [reply api/core/model_runtime/model_providers/tongyi/_common.py]: 😓It's weird. It's been changed

**Review by takatost — APPROVED** [review]
great!

**Pull request merged**